### PR TITLE
[CI/Build] bump ruff version, fix linting issues

### PR DESCRIPTION
**Pull request by dtrifiro**
- bump ruff to 0.5.5
- fix new linting issues
- update ruff workflow for new syntax

### CODE REVIEW

File: vllm/model_executor/layers/quantization/compressed_tensors/compressed_tensors.py [hunk 1]
```diff
@@ -164,7 +164,8 @@ def _is_fp8_w8a8(self, weight_quant: BaseModel,
         is_symmetric_activation = input_quant.symmetric
         is_per_tensor_activation = (
             input_quant.strategy == QuantizationStrategy.TENSOR)
-        if not (is_symmetric_activation and is_per_tensor_activation):
+        if not (is_symmetric_activation  # noqa: SIM103
+                and is_per_tensor_activation):
```
Comment (njhill) [inline vllm/model_executor/layers/quantization/compressed_tensors/compressed_tensors.py:168]: curious of the reason for this change

File: vllm/model_executor/layers/quantization/compressed_tensors/utils.py [hunk 2]
```diff
@@ -80,7 +80,10 @@ class QuantizationArgs(BaseModel):
     )
 
 
-def is_activation_quantization_format(format: str) -> bool:
+def is_activation_quantization_format(format: Union[str, None]) -> bool:
+    if format is None:
+        return False
```
Comment (njhill) [inline vllm/model_executor/layers/quantization/compressed_tensors/utils.py:85]: Why is this needed?

File: vllm/model_executor/layers/quantization/compressed_tensors/utils.py [hunk 3]
```diff
@@ -80,7 +80,10 @@ class QuantizationArgs(BaseModel):
     )
 
 
-def is_activation_quantization_format(format: str) -> bool:
+def is_activation_quantization_format(format: Union[str, None]) -> bool:
```
Comment (njhill) [inline vllm/model_executor/layers/quantization/compressed_tensors/utils.py:83]:
```suggestion
def is_activation_quantization_format(format: Optional[str]) -> bool:
```

File: vllm/model_executor/layers/quantization/compressed_tensors/compressed_tensors.py [hunk 4]
```diff
@@ -66,8 +66,8 @@ def get_quant_method(
     @classmethod
     def from_config(cls, config: Dict[str, Any]) -> "CompressedTensorsConfig":
         target_scheme_map: Dict[str, Any] = dict()
-        ignore: List[str] = config.get("ignore", None)
-        quant_format: str = config.get("format", None)
+        ignore: List[str] = config.get("ignore", [])
+        quant_format: Union[str, None] = config.get("format")
```
Comment (njhill) [inline vllm/model_executor/layers/quantization/compressed_tensors/compressed_tensors.py:70]:
```suggestion
        quant_format: Optional[str] = config.get("format")
```

**Review by njhill — COMMENTED** [review]
Thanks @dtrifiro, the changes look good to me

File: vllm/model_executor/layers/quantization/compressed_tensors/compressed_tensors.py (hunk 1 above)
Comment (dtrifiro) [reply vllm/model_executor/layers/quantization/compressed_tensors/compressed_tensors.py:168]: https://docs.astral.sh/ruff/rules/needless-bool/

Basically `ruff` here tells you that you can just do 

```python
return not(is_symmetric_activation and is_per_tensor_activation)
```

but since we have a comment below, I think it's clearer if we just suppress this error and keep it as is

File: vllm/model_executor/layers/quantization/compressed_tensors/utils.py (hunk 2 above)
Comment (dtrifiro) [reply vllm/model_executor/layers/quantization/compressed_tensors/utils.py]: Thought to make it explicit, but since `None` is not the in the list below we can also get rid of it. Removed in the last force push

File: vllm/model_executor/layers/quantization/compressed_tensors/utils.py (hunk 3 above)
Comment (dtrifiro) [reply vllm/model_executor/layers/quantization/compressed_tensors/utils.py]: Done

File: vllm/model_executor/layers/quantization/compressed_tensors/compressed_tensors.py (hunk 4 above)
Comment (dtrifiro) [reply vllm/model_executor/layers/quantization/compressed_tensors/compressed_tensors.py:70]: Done

**Comment by dtrifiro** [comment]
CI failures look unrelated?

```AssertionError: Error in memory profiling. This happens when the GPU memory was not properly cleaned up before initializing the vLLM instance.```


```[2024-07-24T15:56:13Z] FAILED entrypoints/llm/test_encode.py::test_v1_v2_api_consistency_single_prompt_string[The president of the United States is] - AssertionError: pipeline model parallel group is not initialized
[2024-07-24T15:56:13Z] FAILED entrypoints/llm/test_encode.py::test_v1_v2_api_consistency_single_prompt_string[The capital of France is] - AssertionError: pipeline model parallel group is not initialized
[2024-07-24T15:56:13Z] FAILED entrypoints/llm/test_encode.py::test_v1_v2_api_consistency_single_prompt_string[The future of AI is] - AssertionError: pipeline model parallel group is not initialized
[2024-07-24T15:56:13Z] FAILED entrypoints/llm/test_encode.py::test_v1_v2_api_consistency_single_prompt_tokens[prompt_token_ids0] - AssertionError: pipeline model parallel group is not initialized
[2024-07-24T15:56:13Z] FAILED entrypoints/llm/test_encode.py::test_v1_v2_api_consistency_single_prompt_tokens[prompt_token_ids1] - AssertionError: pipeline model parallel group is not initialized
[2024-07-24T15:56:13Z] FAILED entrypoints/llm/test_encode.py::test_v1_v2_api_consistency_single_prompt_tokens[prompt_token_ids2] - AssertionError: pipeline model parallel group is not initialized
[2024-07-24T15:56:13Z] FAILED entrypoints/llm/test_encode.py::test_v1_v2_api_consistency_single_prompt_tokens[prompt_token_ids3] - AssertionError: pipeline model parallel group is not initialized
[2024-07-24T15:56:13Z] FAILED entrypoints/llm/test_encode.py::test_v1_v2_api_consistency_multi_prompt_string - AssertionError: pipeline model parallel group is not initialized
[2024-07-24T15:56:13Z] FAILED entrypoints/llm/test_encode.py::test_v1_v2_api_consistency_multi_prompt_tokens - AssertionError: pipeline model parallel group is not initialized
[2024-07-24T15:56:13Z] FAILED entrypoints/llm/test_encode.py::test_multiple_pooling_params - AssertionError: pipeline model parallel group is not initialized
[2024-07-24T15:56:13Z] FAILED entrypoints/llm/test_generate.py::test_v1_v2_api_consistency_single_prompt_string[The president of the United States is] - AssertionError: pipeline model parallel group is not initialized
[2024-07-24T15:56:13Z] FAILED entrypoints/llm/test_generate.py::test_v1_v2_api_consistency_single_prompt_string[The capital of France is] - AssertionError: pipeline model parallel group is not initialized
[2024-07-24T15:56:13Z] FAILED entrypoints/llm/test_generate.py::test_v1_v2_api_consistency_single_prompt_string[The future of AI is] - AssertionError: pipeline model parallel group is not initialized
[2024-07-24T15:56:13Z] FAILED entrypoints/llm/test_generate.py::test_v1_v2_api_consistency_single_prompt_tokens[prompt_token_ids0] - AssertionError: pipeline model parallel group is not initialized
[2024-07-24T15:56:13Z] FAILED entrypoints/llm/test_generate.py::test_v1_v2_api_consistency_single_prompt_tokens[prompt_token_ids1] - AssertionError: pipeline model parallel group is not initialized
[2024-07-24T15:56:13Z] FAILED entrypoints/llm/test_generate.py::test_v1_v2_api_consistency_single_prompt_tokens[prompt_token_ids2] - AssertionError: pipeline model parallel group is not initialized
[2024-07-24T15:56:13Z] FAILED entrypoints/llm/test_generate.py::test_v1_v2_api_consistency_single_prompt_tokens[prompt_token_ids3] - AssertionError: pipeline model parallel group is not initialized
[2024-07-24T15:56:13Z] FAILED entrypoints/llm/test_generate.py::test_v1_v2_api_consistency_multi_prompt_string - AssertionError: pipeline model parallel group is not initialized
[2024-07-24T15:56:13Z] FAILED entrypoints/llm/test_generate.py::test_v1_v2_api_consistency_multi_prompt_tokens - AssertionError: pipeline model parallel group is not initialized```

**Comment by dtrifiro** [comment]
Closed in favor of https://github.com/vllm-project/vllm/pull/8469
